### PR TITLE
Add most likely pair filtering for ATHLATES output

### PR DESCRIPTION
**Pull request by rleonid**
- Expose these filters as command line options

### CODE REVIEW

File: src/app/hlarp_cli.ml [hunk 1]
```diff
@@ -60,10 +60,25 @@ let () =
...
     let tool = "athlates" in
-    Term.(const (fun dir ->
-            Athlates.scan_directory dir
+    let equal_pairs_flag =
+      Arg.(value
+           & vflag `MostLikelyPair
+              [ `MostLikelyPair, info ["most_likely_pair"]
+                  ~doc:"For each typing file report the most frequent two alleles (default)."
+              ; `ReportAll,      info ["all_pairs"]
+                  ~doc:"For each typing file report all pairs (will create duplicates)."
```
Comment (ihodes) [inline src/app/hlarp_cli.ml]: What is duplicated in requesting all pairs or first pair?

File: src/app/hlarp_cli.ml [hunk 2]
```diff
@@ -60,10 +60,25 @@ let () =
   in
   let athlates =
     let tool = "athlates" in
-    Term.(const (fun dir ->
-            Athlates.scan_directory dir
+    let equal_pairs_flag =
+      Arg.(value
+           & vflag `MostLikelyPair
+              [ `MostLikelyPair, info ["most_likely_pair"]
+                  ~doc:"For each typing file report the most frequent two alleles (default)."
```
Comment (ihodes) [inline src/app/hlarp_cli.ml]: ITOCTYDK (in the off-chance that you don't know) FWIW  Cmdliner will already note that the `MostLikelyPair is the default (by saying absent=most_likely_pair, which isn't all that great, so the additional note that it's the default in the docstring is still nice).

File: src/lib/hlarp.ml [hunk 3]
```diff
@@ -175,34 +176,46 @@ module Athlates = struct
...
         close_in ic;
         acc
     in
     let alleles = loop false [] in
+    let alleles =
+      match equal_pairs with
+      | `ReportAll      -> alleles
+                        (* Will fail silently if < 2, I think ok. *)
+      | `FirstPair      -> List.take (List.rev alleles) 2
+      | `Unique         -> List.dedup alleles
```
Comment (ihodes) [inline src/lib/hlarp.ml]: Could this also fail silently?

Comment (rleonid) [inline src/lib/hlarp.ml]: By fail it will just return an empty list -> not a hard crash.

File: src/app/hlarp_cli.ml (hunk 1 above)
Comment (rleonid) [inline src/app/hlarp_cli.ml]: An allele might be part of more than one pair, so you'll see it repeated.

File: src/app/hlarp_cli.ml (hunk 2 above)
Comment (rleonid) [inline src/app/hlarp_cli.ml]: I did not know or notice that, thank you!

**Pull request merged**